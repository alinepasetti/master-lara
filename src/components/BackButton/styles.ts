import styled, { css } from 'styled-components';
import { BiArrowBack } from 'react-icons/bi';

export const Back = styled(BiArrowBack)`${({ theme }) => css`
  width: 40px;
  height: 40px;
  color: ${theme.colors.mediumGray};
  background-color: ${theme.colors.white};
  font-size: 0.8rem;
  border-radius: 50%;
  padding: 0.7rem;
  position: absolute;
  left: 1.6rem;
  top: 1.6rem;
`}`;
