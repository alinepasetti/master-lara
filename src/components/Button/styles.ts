import styled, { css } from 'styled-components';

export const Button = styled.a` ${({ theme }) => css`
  width: 90%;
  display: block;
  padding: 1.6rem;
  font-size: 2.4rem;
  background: ${theme.colors.primaryColor};
  color: ${theme.colors.white};
  border: none;
  margin: 0 auto 2.4rem;
  text-align: center;
  text-decoration: none;
`}`;
