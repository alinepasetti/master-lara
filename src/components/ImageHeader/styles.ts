import styled from 'styled-components';

export const Image = styled.img`
  width: 100%;
  height: 40vh;
  object-fit: cover;
  object-position: center;
`;
