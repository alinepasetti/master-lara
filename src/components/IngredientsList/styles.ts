import styled, { css } from 'styled-components';
import { AiFillCloseCircle } from 'react-icons/ai';
import { AiFillCheckCircle } from 'react-icons/ai';

export const Container = styled.ul`
  padding: 2.4rem;
  padding-top: 0;
  margin: 0;
  > a {
    margin-top: 1.6rem;
  }
`;

export const ListItem = styled.li`
  list-style: none;
  display: flex;
  align-items: center;
`;

export const CheckIcon = styled(AiFillCheckCircle)`${({ theme }) => css`
  color: ${theme.colors.green};
  margin-right: 2.4rem;
  height: 2.4rem;
  width: 2.4rem;
`}`;

export const UncheckIcon = styled(AiFillCloseCircle)`${({ theme }) => css`
  color: ${theme.colors.secondaryColor};
  margin-right: 2.4rem;
  height: 2.4rem;
  width: 2.4rem;
`}`;
