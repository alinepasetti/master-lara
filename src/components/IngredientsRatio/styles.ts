import styled, { css } from 'styled-components';

export const Container = styled.p`${({ theme }) => css`
  font-size: 1.6rem;
  margin: 0 0;
  color: ${theme.colors.white};
  background-color: ${theme.colors.primaryColor};
  font-size: 1.2rem;
  font-weight: 400;
  position: absolute;
  right: 1.6rem;
  top: 1.6rem;
  border-radius: 8px;
  box-shadow: 0 0 5px ${theme.colors.mediumGray};
  padding: 1rem;
`}`;
