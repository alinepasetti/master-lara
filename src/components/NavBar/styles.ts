import styled from 'styled-components';

export const NavBar = styled.nav`
  width: 100%;
  padding: 1rem 1.6rem;
  background-color: black;
`;

export const Image = styled.img`
  width: 20%;
  display: block;
  margin: 0 auto;
`;
