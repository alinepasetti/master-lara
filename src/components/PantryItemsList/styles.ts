import styled, { css } from 'styled-components';

type HasIngredient = {
  hasIngredient?: boolean;
};

export const Container = styled.ul`
  height: 100%;
  width: 100%;
  min-height: 4.0rem;
  padding: 0 0.8rem;
  margin: 1.6rem 0;
  display: flex;
  justify-content: space-around;
  flex: 1;
  flex-wrap: wrap;
  list-style: none;
`;

export const PantryItem = styled.li<HasIngredient>`${({ theme }) => css`
  color: ${theme.colors.primaryColor};
  background-color: ${theme.colors.white};
  padding: 1.25rem;
  width: 47%;
  border-radius: 8px;
  margin: 0.8rem 0;
  & > span {
    display: flex;
    justify-content: space-between;
  }

  & > span > svg {
    width: 2.5rem;
  }

  @media screen and (min-width: 768px) {
    width: 31.5%;
  }

  @media screen and (min-width: 1164px) {
    width: 23%;
  }
`}`;

export const Message = styled.p`${({ theme }) => css`
  font-size: 1.6rem;
  width: 100%;
  text-align: center;
  margin: 0.8rem 0;
  color: ${theme.colors.primaryColor};
`}`;
