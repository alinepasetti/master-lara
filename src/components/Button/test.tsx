import { screen } from '@testing-library/react';
import { Button } from '.';
import { renderTheme } from '../../styles/render-theme';
import { vi } from 'vitest';

describe('<Button />', () => {
  it('should render a button', () => {
    renderTheme(<Button clickHandler={() => 'hi'}>Example</Button>);
    const button = screen.getByText('Example');
    expect(button).toBeInTheDocument();
  });

  it('should call the click handler', () => {
    const clickHandler = vi.fn();
    renderTheme(<Button clickHandler={clickHandler}>Example</Button>);
    const button = screen.getByText('Example');
    button.click();
    expect(clickHandler).toBeCalledTimes(1);
  });

  it('should match snapshot', () => {
    renderTheme(<Button clickHandler={() => 'hi'}>Example</Button>);
    expect(screen.getByText('Example')).toMatchSnapshot();
  });
});
