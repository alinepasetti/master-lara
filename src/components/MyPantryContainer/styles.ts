import styled from 'styled-components';
import { Button as SearchButton } from '../Button/styles';

export const Container = styled.div`
  width: 100%;
  display: flex;
  flex-direction: column;
  align-items: center;

  ${SearchButton} {
    border-radius: 8px;
    width: 10.8rem;
  }
`;
