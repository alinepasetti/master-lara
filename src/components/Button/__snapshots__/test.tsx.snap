// Vitest Snapshot v1

exports[`<Button /> > should match snapshot 1`] = `
.c0 {
  padding: 1.6rem;
  font-size: 1.6rem;
  font-weight: 700;
  background: #16426cff;
  color: #FFFFFF;
  border: none;
  margin: 0 auto;
  -webkit-letter-spacing: 0.5px;
  -moz-letter-spacing: 0.5px;
  -ms-letter-spacing: 0.5px;
  letter-spacing: 0.5px;
}

<button
  class="c0"
>
  Example
</button>
`;
